File: stress/t/arr_boollogic.f
      PROGRAM ARRBOOLLOGIC
      INTEGER*8 N,I,CNT,BOTH
      LOGICAL TB,UB,ALLF,ANYT,ALT
      LOGICAL B(0:23),C(0:23)
      N=24
      I=0
   10 IF (I.GE.N) GOTO 20
      B(I)=(MOD(I,2_8).EQ.0)
      C(I)=(MOD(I,3_8).EQ.0)
      I=I+1
      GOTO 10
   20 CONTINUE
      CNT=0
      I=0
   30 IF (I.GE.N) GOTO 40
      TB=B(I)
      IF (TB) CNT=CNT+1
      I=I+1
      GOTO 30
   40 CONTINUE
      ALLF=.TRUE.
      I=0
   50 IF (I.GE.N) GOTO 60
      TB=B(I)
      ALLF=ALLF.AND.TB
      I=I+1
      GOTO 50
   60 CONTINUE
      ANYT=.FALSE.
      I=0
   70 IF (I.GE.N) GOTO 80
      TB=C(I)
      ANYT=ANYT.OR.TB
      I=I+1
      GOTO 70
   80 CONTINUE
      BOTH=0
      I=0
   90 IF (I.GE.N) GOTO 100
      TB=B(I)
      UB=C(I)
      IF (TB.AND.UB) BOTH=BOTH+1
      I=I+1
      GOTO 90
  100 CONTINUE
      ALT=.TRUE.
      I=0
  110 IF (I.GE.N-1) GOTO 120
      TB=B(I)
      UB=B(I+1)
      IF (TB.EQV.UB) ALT=.FALSE.
      I=I+1
      GOTO 110
  120 CONTINUE
      WRITE(*,'(A,I0)') 'count_true_B=',CNT
      IF (ALLF) THEN
        WRITE(*,'(A)') 'all_B=true'
      ELSE
        WRITE(*,'(A)') 'all_B=false'
      ENDIF
      IF (ANYT) THEN
        WRITE(*,'(A)') 'any_C=true'
      ELSE
        WRITE(*,'(A)') 'any_C=false'
      ENDIF
      WRITE(*,'(A,I0)') 'both_BC=',BOTH
      IF (ALT) THEN
        WRITE(*,'(A)') 'alternating_B=true'
      ELSE
        WRITE(*,'(A)') 'alternating_B=false'
      ENDIF
      END

File: stress/t/arr_boollogic.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t n,i,cnt,both;
  int tb,ub,allf,anyt,alt;
  int B[24],C[24];
  n=24;
  for(i=0;i<n;i++){ B[i]=(i%2==0); C[i]=(i%3==0); }
  cnt=0;
  for(i=0;i<n;i++){ tb=B[i]; if(tb){ cnt=cnt+1; } }
  allf=1;
  for(i=0;i<n;i++){ tb=B[i]; allf=allf&&tb; }
  anyt=0;
  for(i=0;i<n;i++){ tb=C[i]; anyt=anyt||tb; }
  both=0;
  for(i=0;i<n;i++){ tb=B[i]; ub=C[i]; if(tb&&ub){ both=both+1; } }
  alt=1;
  for(i=0;i<n-1;i++){ tb=B[i]; ub=B[i+1]; if(tb==ub){ alt=0; } }
  printf("count_true_B="); printf("%ld",(long)cnt); printf("\n");
  printf("all_B="); printf("%s", allf?"true":"false"); printf("\n");
  printf("any_C="); printf("%s", anyt?"true":"false"); printf("\n");
  printf("both_BC="); printf("%ld",(long)both); printf("\n");
  printf("alternating_B="); printf("%s", alt?"true":"false"); printf("\n");
  return 0;
}
